### DISCUSSION TITLE
Rotate column labels

### DISCUSSION THREAD

**Discussion by david-waterworth** [posit-dev/great-tables#412] [Q&A] [answered]
Is it possible to rotate column labels by 90 degrees? I have a large pivot table and both the row and col labels are longish (and the values a single digit counts or "-" (Null) so to make it somewhat more readable I'd like to rotate the column labels?

**Comment by jrycw** [comment, marked as answer]
@david-waterworth , you could try adding some custom CSS like the code below:
```python
import polars as pl
from great_tables import GT, html


(
    GT(pl.DataFrame({"longish": [1, 2]}))
    .cols_label(longish=html('<h3 style="transform: rotate(90deg);">longish</h3>'))
    
)
```
![image](https://github.com/user-attachments/assets/9116f85b-815d-40b6-9964-815877d3cbb7)

**Reply by travis-leith** [reply]
does this have to be done for all column headers individually, or can I do it for all headers in one go? In R, this is what I do:

```R
gt_obj |>
    opt_css(
      css = "
    #constraints_display .gt_col_heading {
    writing-mode: vertical-rl;
    transform: scale(-1);
    vertical-align: middle;
    font-weight: bold;
    }")
```

**Reply by travis-leith** [reply]
also, this solution does not work for really long column names:

```python
(
    GT(pd.DataFrame({"loooooooooooooongish": [1, 2]}))
    .cols_label(loooooooooooooongish=html('<h3 style="transform: rotate(90deg);">loooooooooooooongish</h3>'))
    
)
```

![image](https://github.com/user-attachments/assets/3ba65e49-452c-4084-8e54-321b5c5f7b7b)

**Reply by jrycw** [reply]
Oops, I'm not sure if [style.css()](https://posit-dev.github.io/great-tables/blog/introduction-0.13.0/index.html#using-custom-style-rules-with-the-new-style.css) would be helpful in this case.

**Reply by travis-leith** [reply]
booyah!

```python
(
    GT(pd.DataFrame({"loooooooooooooongish": [1, 2]}))
    .tab_style(
        style = style.css(rule = "writing-mode: vertical-rl;transform: scale(-1);vertical-align: middle;font-weight: bold;"),
        locations = loc.column_header()
    )
)
```

![image](https://github.com/user-attachments/assets/f37523db-ec9d-4947-be6e-83b0aa4492f4)

**Reply by travis-leith** [reply]
this also answer my earlier question about having it apply to all columns

**Reply by jrycw** [reply]
@travis-leith, thanks for sharing your solution—`.cols_label_rotate()` has just been merged, inspired by your approach!

**Comment by jrycw** [comment]
It seems there are no follow-up issues related to this discussion, so I'll close it for now.
